src/main.rs

```diff
@@ -289,7 +289,24 @@ fn main_try() -> Result<()> {
 
     log::info!("Protocol speed {} kHz", protocol_speed);
 
-    let mut session = probe.attach(chip).context("failed attaching to target")?;
+    let mut session = if config.general.connect_under_reset {
+        probe
+            .attach_under_reset(chip)
+            .context("failed attaching to target")?
+    } else {
+        let potential_session = probe.attach(chip);
+        match potential_session {
+            Ok(session) => session,
+            Err(session) => {
+                log::info!("The target seems to be unable to be attached to.");
+                log::info!(
+                    "A hard reset during attaching might help. This will reset the entire chip."
+                );
+                log::info!("Set `general.connect_under_reset` in your cargo-embed configuration file to enable this feature.");
+                return Err(session).context("failed attaching to target");
```

Comment:
```suggestion
            Err(err) => {
                log::info!("The target seems to be unable to be attached to.");
                log::info!(
                    "A hard reset during attaching might help. This will reset the entire chip."
                );
                log::info!("Set `general.connect_under_reset` in your cargo-embed configuration file to enable this feature.");
                return Err(err).context("failed attaching to target");
```

Naming the error `session` is a bit confusing to me.